src/common/contentPreference.ts

```diff
@@ -516,3 +518,28 @@ export const unblockEntity = async ({
   // currently unblock is just like unfollow, eg. remove everything from db
   return unfollowEntity({ ctx, id, entity, feedId });
 };
+
+export const whereNotUserBlocked = (
+  qb: QueryBuilder,
+  {
+    userId,
+    feedId,
+  }: Partial<{
+    userId: string;
+    feedId: string;
+  }>,
+) => {
+  const feedIds = feedId ? [feedId, userId] : [userId];
+  const query = qb
+    .subQuery()
+    .from(ContentPreference, 'pref')
+    .where({
+      status: ContentPreferenceStatus.Blocked,
+      type: ContentPreferenceType.User,
+      userId,
+      feedId: In(feedIds),
+    })
+    .andWhere(`"pref"."referenceId" = ${qb.alias}."userId"`)
+    .getQuery();
```

Comment:
content preference is a big table, can you show examples of query plans for this to make sure we have indexes at least, on top of my head not sure if there is a more efficient way to query this.

Reply:
We need always to search for status/userId/feedId/referenceId, maybe we can remove type but I don't thing is to much impactful. We don't have indexes that involve all those field at the moment. Should I add one? 🤔

Reply:
we do have `idx_content_preferences_feedid_type_userid_status` and the rest is PK, as said can you check the query plan to see how it looks and if we need to add anything?

Reply:
Seems that we don't have any use of indexes with this query, I've tried to add `.useIndex()` but didn't work. I'm going to test if adding an index for all the fields changes something.

<img width="1094" alt="Pasted Graphic" src="https://github.com/user-attachments/assets/af5f64b6-2e60-4385-a04e-16d205f4dcec" />

Reply:
did you do it locally or in prod, locally psql will almost never use indexes because there is not enough data?

basically best way is to test on metabase or read replica, we can take a look on monday together if you need help